src/Services/HubspotCommerce.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Drupal\hubspot_commerce\Services;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\hubspot_api\Manager;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Exception;
+
+/**
+ * The HubspotCommerce service class.
+ *
+ * Contains functions to sync data with the Hubspot API.
+ *
+ * @package Drupal\hubspot_commerce
+ */
+class HubspotCommerce {
+
+  /**
+   * The Hubspot API Manager.
+   *
+   * @var \Drupal\hubspot_api\Manager
+   */
+  protected $hubspotManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Logger service.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * The client.
+   *
+   * @var \SevenShores\Hubspot\Http\Client
+   */
+  public $client;
+
+  /**
+   * Constructs a new HubSpot Commerce service instance.
+   *
+   * @param \Drupal\hubspot_api\Manager $hubspot_manager
+   *   The Hubspot API Manager class.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manger.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The logger factory.
+   *
+   * @throws \Exception
+   */
+  public function __construct(
+    Manager $hubspot_manager,
+    EntityTypeManagerInterface $entity_type_manager,
+    LoggerChannelFactoryInterface $logger_factory
+  ) {
+    $this->hubspotManager = $hubspot_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logger = $logger_factory->get(HUBSPOT_COMMERCE_LOGGER_CHANNEL);
+
+    // Initialize our Hubspot API client.
+    try {
+      $this->client = $this->hubspotManager->getHandler()->client;
+    }
+    catch (Exception $e) {
+      $this->logger->error('Could not create a Hubspot API client. Error: @error', [
+        '@error' => $e->getMessage(),
+      ]);
+    }
+  }
+
+  /**
+   * Handles the queue items and syncs them with the Hubspot API.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity we're syncing (ie. user/order/product variation).
+   */
+  public function syncQueueItem(EntityInterface $entity) {
+    // TODO: Depending on which entity we're trying to sync, trigger the
+    // appropriate action.
+  }
```

Comment:
**CHANGE REQUEST**

*Readability*

Nothing specific to queue items here, this service generally syncs an entity and it could be called from anywhere in the code including from other modules. Let's rename this function to just `sync`.

**CHANGE REQUEST**

*Architecture*

Let's define the architecture here.

* An entity comes in for syncing.
* An event is dispatched (`commerce_hubspot.entity_sync_to.entity_mapping`) that allows modules to determine the HubSpot entity type (e.g. `contact` or `deal`) and HubSpot entity ID that the Drupal entity will be synced to.
* If the final response does not contain both (type and ID) we consider that the entity is not to be synced.
* If we do have a good response, we dispatch another event  (`commerce_hubspot.entity_sync_to.field_mapping`) allowing modules to define which Drupal fields will be synced to which HubSpot fields.
* If the response is an empty array, no fields to sync.
* If there are fields to sync, we call another internal function that prepares the payload for syncing the properties; it should be the same for all HubSpot entity types.
* We then call the right function depending on the HubSpot entity type, either through protected functions defined per type here as you currently have it or directly through the PHP SDK if there are no other differences.

We then need to create the following event subscribers in this module:
* `UserSubscriber`
* `CommerceOrderSubscriber`
* `CommerceOrderItemSubscriber`
* `CommerceProductVariationSubscriber`
all responding to both events above (`entityMapping` and `fieldMapping` functions).

Reply:
I think I'll break this down into smaller features that will be part of issues #3, #4, and #5.